ocaml/typing/typecore.ml

```diff
@@ -7780,19 +7771,21 @@ and type_construct env (expected_mode : expected_mode) loc lid sarg
     | None -> []
     | Some se -> begin
         match Jane_syntax.Expression.of_ast se with
-        | Some (Jexp_tuple (_ : _ list), _) when
-            constr.cstr_arity > 1 || Builtin_attributes.explicit_arity attrs ->
-          raise(Error(loc, env, Constructor_labeled_arg))
-        | Some (( Jexp_tuple _
-                | Jexp_comprehension _
+        | Some (( Jexp_comprehension _
                 | Jexp_immutable_array _
                 | Jexp_n_ary_function _
                 | Jexp_layout _
                 | Jexp_modes _ ), _) -> [se]
         | None -> match se.pexp_desc with
         | Pexp_tuple sel when
             constr.cstr_arity > 1 || Builtin_attributes.explicit_arity attrs
-          -> sel
+          ->
+          if
+            List.exists (function Some _, _ -> true | _ -> false) sel
```

Comment:
This little line is repeated several times. Maybe name it and make a new function?